include/GenSync/Syncs/BloomFilter.h

```diff
@@ -0,0 +1,231 @@
+     */
+    class Builder
+    {
+    public:
+        /** Constructor for builder pattern */
+        Builder(){
+            bfSize = 0;
+            numHashes = 0;
+            numExpElems = 0;
+            falsePosProb = 0;
+        }
+
+        /**
+         * Builds a BloomFilter object.
+         * setSize and setNumHashes are meant to be used together.
+         * setNumExpElems and setFalsePosProb are meant to be used together.
+         * If there is a mismatched use of setters, output error message and quit, returnng empty BloomFilter.
+         * @return a BloomFilter object from the build parts that have been set.
+         */    
+        BloomFilter build()
+        {
+            if(bfSize != 0 && numHashes != 0)
+                return BloomFilter(bfSize, numHashes);
+            
+            if(numExpElems != 0 && falsePosProb != 0)
+                return BloomFilter(numExpElems, falsePosProb);
+            
+            Logger::error_and_quit("ERROR: Mismatched combination of setters used in BloomFilter construction!");
+            return BloomFilter();
+        }        
+
+        /**
+         * Sets the size of the BloomFilter, specifically its length in bits.
+         * Meant to be used with setNumHashes setter. Cannot be used with other setters.
+         * Discards all elements inserted before the resize.
+         */
+        Builder& setSize(size_t size)
+        {
+            bfSize = size;
+            return *this;
+        }
+
+        /**
+         * Sets the number of hash functions used by the BloomFilter for each element insertion
+         * Meant to be used with setSize setter. Cannot be used with other setters.
+         * Discards all elements inserted before the hashes reset.
+         */
+        Builder& setNumHashes(int nHash)
+        {
+            numHashes = nHash;
+            return *this;
+        }
+
+        /**
+         * Sets the number of expected elements in the BloomFilter.
+         * Meant to be used with setFalsePosProb setter. Cannot be used with other setters.
+         * Discards all elements inserted before the resize.
+         */
+        Builder& setNumExpElems(size_t numElems)
+        {
+            numExpElems = numElems;
+            return *this;
+        }
+
+        /**
+         * Sets the probability of false positives in the BloomFilter
+         * Meant to be used with setNumExpElems setter. Cannot be used with other setters.
+         * Discards all elements inserted before reset.
+         */
+        Builder& setFalsePosProb(float prob)
+        {
+            falsePosProb = prob;
+            return *this;
+        }
+        
+    private:
+        // Length of the BloomFilter in bits
+        size_t bfSize;
+
+        // Number of hash functions used by the BloomFilter
+        int numHashes;
+
+        // Number of expected elements in BloomFilter
+        size_t numExpElems;
+
+        // The probability/rate of false positives

```

Comment:
This is an approximate probability of false positives. I added to the documentation to specify that it is an approximation rather than an upper bound.

Reply:
Ok ...